### ISSUE TITLE
Heroic Cannot Install Game from Epic Games (Install Finished)

### ISSUE THREAD

**Issue by puddingconfident45** [Heroic-Games-Launcher/HeroicGamesLauncher#2275]
### Describe the bug

I've been trying to install Far Cry Primal on Steam Deck via Heroic but the installation will start and then immediately complete with the game not getting actually installed.

It doesn't give any error message.

This happens both on internal and external drives.

Worth to mention is that after purchasing the game, I was actually able to Install it via Heroic. Due to poor performance I then uninstalled it and installed it via Epic Games Launcher. Now that I switched back to Heroic, it doesn't seem to want to install anymore.

So far I've encountered this issue only with this specific game.

### Add logs

```text
(10:56:26) INFO: [Connection]: Connectivity: check-online
(10:56:26) INFO: [Connection]: Pinging external endpoints
(10:56:26) INFO: [Legendary]: Running command: /app/bin/heroic/resources/app.asar.unpacked/
build/bin/linux/legendary --version
(10:56:26) INFO: [Legendary]: Legendary location: /app/bin/heroic/resources/
app.asar.unpacked/build/bin/linux/legendary
(10:56:26) INFO: [Gog]: GOGDL location: /app/bin/heroic/resources/app.asar.unpacked/
build/bin/linux/gogdl
(10:56:26) WARNING: [Backend]: Failed to register protocol with OS.
(10:56:26) INFO: [Connection]: Connectivity: online
(10:56:26) INFO: [Backend]: AreWeAntiCheatYet data downloaded
(10:56:26) INFO: [Gog]: Running command: /app/bin/heroic/resources/app.asar.unpacked/
build/bin/linux/gogdl --version
(10:56:27) INFO: [Backend]:
Heroic Version: 2.5.2 Yamato
Legendary Version: 0.20.31 Dark Energy (hotfix #5)
GOGdl Version: 0.5
OS: Freedesktop SDK KERNEL: 5.13.0-valve36-1-neptune ARCH: x64
CPU: AMD Custom APU 0405 @2.29 GOVERNOR: schedutil
RAM: Total: 11.52 GiB Available: 8.47 GiB
GRAPHICS: GPU0: VanGogh [AMD Custom GPU 0405] VRAM: 512MB
PROTOCOL: x11
(10:56:27) INFO: [Backend]: Checking for current version changelog
(10:56:27) INFO: [Frontend]: Refreshing Library
(10:56:27) INFO: [Legendary]: Refreshing library...
(10:56:27) INFO: [Legendary]: Refreshing Epic Games...
(10:56:27) INFO: [Legendary]: Game list updated, got 34 games & DLCs
(10:56:27) INFO: [Backend]: Frontend Ready
(10:56:27) INFO: [Frontend]: Force Update
(10:56:28) INFO: [Legendary]: Running command: /app/bin/heroic/resources/app.asar.unpacked/
build/bin/linux/legendary list --third-party
(10:56:28) INFO: [Winetricks]: Downloading Winetricks
(10:56:30) INFO: [Backend]: Using cached HowLongToBeat data for Far Cry Primal
(10:56:30) DEBUG: [Legendary]: Using cached install info
(10:56:30) DEBUG: [Legendary]: Using cached install info
(10:56:31) DEBUG: [Legendary]: Using cached install info
(10:56:31) DEBUG: [Backend]: {"free":6183645184,"diskSize":6183772160,"message":"5.76 GiB
/ 5.76 GiB","validPath":true}
(10:56:31) DEBUG: [Backend]: {"free":6183645184,"diskSize":6183772160,"message":"5.76 GiB
/ 5.76 GiB","validPath":true}
(10:56:32) INFO: [Backend]: Starting the Download Queue
(10:56:54) DEBUG: [Backend]: {"free":93849600000,"diskSize":502683049984,"message":"87.4
GiB / 468.16 GiB","validPath":true}
(10:57:00) INFO: [Backend]: Writing config for Larkspur
(10:57:00) INFO: [DownloadManager]: Larkspur added to download manager queue.
(10:57:00) DEBUG: [Legendary]: Using cached install info
(10:57:00) INFO: [Backend]: Preventing machine to sleep
(10:57:00) DEBUG: [Legendary]: Using cached install info
(10:57:00) DEBUG: [Legendary]: Using cached install info
(10:57:01) DEBUG: [Legendary]: Using cached install info
(10:57:01) INFO: [Legendary]: Installing Larkspur: /app/bin/heroic/resources/
app.asar.unpacked/build/bin/linux/legendary install Larkspur --platform Windows --base-path /run/
media/mmcblk0p1/Games/Heroic --skip-dlcs --skip-sdl -y
(10:57:01) DEBUG: [Legendary]: Logging to file "/home/deck/.var/app/
com.heroicgameslauncher.hgl/config/heroic/GamesConfig/Larkspur.log"
(10:57:04) INFO: [Backend]: Adding shortcuts for Far Cry Primal
(10:57:04) INFO: [DownloadManager]: Finished installing of Larkspur
(10:57:04) INFO: [DownloadManager]: Larkspur added to download manager finished.
(10:57:04) INFO: [DownloadManager]: Larkspur removed from download manager.
(10:57:04) INFO: [Frontend]: Refreshing Library
(10:57:04) INFO: [Legendary]: Refreshing library...
(10:57:04) INFO: [Legendary]: Game list updated, got 34 games & DLCs
(10:57:04) DEBUG: [Legendary]: Using cached install info
(10:57:04) DEBUG: [Legendary]: Using cached install info
(10:57:04) DEBUG: [Legendary]: Using cached install info
(10:57:04) DEBUG: [Legendary]: Using cached install info
(10:57:04) DEBUG: [Legendary]: Using cached install info
(11:00:24) INFO: [Backend]: Using cached HowLongToBeat data for Far Cry Primal
(11:00:24) DEBUG: [Legendary]: Using cached install info
(11:00:24) DEBUG: [Legendary]: Using cached install info
(11:00:54) INFO: [Backend]: Using cached HowLongToBeat data for Far Cry Primal
(11:00:54) DEBUG: [Legendary]: Using cached install info
(11:00:54) DEBUG: [Legendary]: Using cached install info
```

### Steps to reproduce

1. With Epic account connected, go to "library"
2. Click on Far Cry Primal
3. Install Game on internal or external drive

### Expected behavior

Being able to install the game.

### Screenshots

![Screenshot_20221228_110004](https://user-images.githubusercontent.com/121291037/209797520-c3c9126a-1a96-46a1-8b77-8e62ff935eaf.png)
![Screenshot_20221228_110317](https://user-images.githubusercontent.com/121291037/209797528-7d317ac7-e5ab-465b-ba7f-6ca79a348df9.png)

### Heroic Version

Latest Stable

### System Information

- OS: SteamOS (Steam Deck)

### Additional information

This is not a "new" error. It happened for months but I only re-tried it now and never reported before.

**Comment by Nocccer** [comment]
Can you delete `~/Games/Heroic/FarCry` folder?

**Comment by puddingconfident45** [comment]
> Can you delete `~/Games/Heroic/FarCry` folder?
> 
> Btw once activate via `legendary activate --uplay` you also can install the game from the uplay launcher. No epic needed.

Thanks. I don't have any FarCry folder in the internal or external drive. 

How do I activate that?

**Comment by MrRubberDucky** [comment]
Same happening here on Windows 10, takes long to fetch size then instantly finishes it after a minute. No game folders created.

**Comment by arielj** [comment]
Some Ubisoft games cannot be installed using Heroic directly (same happens with Watchdogs 2, I imagine the same is happening for Far Cry Primal)

You need to install the Ubisoft client for those games (option 1 here https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/wiki/How-to-install-Ubisoft-Connect-on-Linux-and-Mac)

This seems to be a limitation of Legendary (the tool we use to download Epic games) but I understand it's not Legendary's fault but it's just that Epic doesn't provide downloadable files.

Try running this command `/app/bin/heroic/resources/
app.asar.unpacked/build/bin/linux/legendary install Larkspur --platform Windows --base-path /run/
media/mmcblk0p1/Games/Heroic --skip-dlcs --skip-sdl -y` and check the output, at least for Watchdogs 2 it says that the game cannot be installed using Legendary

**Comment by arielj** [comment]
@puddingconfident45  @MrRubberDucky can you confirm if what I mentioned in the previous comment is what's happening to you? I don't own Far Cry Primal to test it.

**Comment by MrRubberDucky** [comment]
> @puddingconfident45 @MrRubberDucky can you confirm if what I mentioned in the previous comment is what's happening to you? I don't own Far Cry Primal to test it.

For me it happens with any game so I moved to EGL for now.

**Comment by arielj** [comment]
Can anyone confirm if this is still happening? and if it is, can you share the legendary log when an installation fails? (settings > logs > legendary/epic)

**Comment by jveldboom** [comment]
We ran into a similar issue where when we clicked on download, a notification would immediately popup saying it was starting and then finished. But the game was not installed. For us, restarting the Steam Deck resolved the issue.

I believe the root cause was not letting a download complete. We switched to Steam Deck game mode while an active download was in progress. Hope this helps someone and am happy to provide more details.

**Comment by arielj** [comment]
Closing, if the original issue is what I think it is, it should be fixed in 2.15.0

For anyone else having installation issues it would be better to create another issue since it's probably a different problem.

**Comment by ShadowJonathan** [comment]
Im seeing the same issue: I was able to get a download of Alan Wake 2 started, but after I aborted it and went back to gaming mode for a bit, and then back to desktop mode, I cannot install anything without it immediately "completing"

I've tried resetting and logging back in, but none of that fixes it it seems
